lldb/source/API/SBThread.cpp

```diff
@@ -713,7 +731,10 @@ void SBThread::StepInstruction(bool step_over, SBError &error) {
   LLDB_INSTRUMENT_VA(this, step_over, error);
 
   std::unique_lock<std::recursive_mutex> lock;
-  ExecutionContext exe_ctx(m_opaque_sp.get(), lock);
+  Process::StopLocker stop_locker;
+  ExecutionContext exe_ctx(m_opaque_sp.get(), lock, stop_locker);
+  if (!stop_locker.IsLocked())
+    return;
```

Comment:
And this one.  
Maybe we should make the ExecutionContext constructor take an optional Status * object, and if that's not empty we can put a standard error in it?  Then you could just pass in these errors, and then you wouldn't have to set the error directly each time.
You could also use that for your SBValue error return; I'm pretty sure there's a way to make a ValueObject just from a Status.